shell/platform/darwin/common/framework/Source/FlutterBaseDartProject.mm

```diff
@@ -0,0 +1,89 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/common/framework/Headers/FlutterBaseDartProject.h"
+#import "flutter/shell/platform/darwin/common/framework/Source/FlutterBaseDartProject_Internal.h"
+
+#include <vector>
+
+#include "flutter/shell/platform/common/engine_switches.h"
+
+static NSString* const kICUBundlePath = @"icudtl.dat";
+static NSString* const kAppBundleIdentifier = @"io.flutter.flutter.app";
+
+@implementation FlutterBaseDartProject {
+  NSBundle* _dartBundle;
+  NSString* _assetsPath;
+  NSString* _ICUDataPath;
+}
+
+- (instancetype)init {
+  return [self initWithPrecompiledDartBundle:nil];
+}
+
+- (instancetype)initWithPrecompiledDartBundle:(NSBundle*)bundle {
+  self = [super init];
+  NSAssert(self, @"Super init cannot be nil");
+
+  _dartBundle = bundle ?: [NSBundle bundleWithIdentifier:kAppBundleIdentifier];
```

Comment:
I was surprised to see we're using a language extension here, but looks like the style guide specifically carves out an exception for it here:
https://google.github.io/styleguide/objcguide.html#nonstandard-extensions